code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+	enraged.cut_overlay(hate_fire)
+
+///destroying the rune will also remove the punch force of the persons on the rune.
+/obj/effect/warped_rune/redspace/Destroy()
+	for(var/mob/living/carbon/human/enraged in rune_turf)
+		UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+		enraged.cut_overlay(hate_fire)
+	return ..()
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	max_cooldown = 100
+	///path of the xeno resin sheet to spawn by the rune
+	var/obj/item/stack/sheet/xeno_resin/resin
+
+
+/obj/item/stack/sheet/xeno_resin
+	name = "Resin sheets"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "nestoverlay" //literally just the xeno nest icon
+	merge_type = /obj/item/stack/sheet/xeno_resin
+
+
+/obj/effect/warped_rune/greenspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greenspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_resin()
+
+
+/obj/effect/warped_rune/greenspace/proc/transmute_resin()
+	for(var/obj/item/stack/sheet/mineral/plasma/plasma_sheet in rune_turf)
+		resin = new(rune_turf)
+		plasma_sheet.amount--
+		if(plasma_sheet.amount <= 0)
+			qdel(plasma_sheet)
+
+
+//note : some of these can only be built ON resin weeds such as the resin nest.
+GLOBAL_LIST_INIT(resin_recipes, list ( \
+	new/datum/stack_recipe("Resin seed", /obj/structure/alien/weeds/node, 10, one_per_turf = 1, on_floor = 1, time = 100), \
+	new/datum/stack_recipe("Resin Wall", /obj/structure/alien/resin/wall, 3, one_per_turf = 1, on_floor = 1 , time = 50), \
+	new/datum/stack_recipe("Fake hatched egg", /obj/structure/alien/egg/burst, 2, one_per_turf = 1, on_floor = 1), \
+	new/datum/stack_recipe("Resin nest", /obj/structure/bed/nest , 5, one_per_turf = 1, on_floor = 0, time = 50), \
+	))
+
+
+/obj/item/stack/sheet/xeno_resin/get_main_recipes()
+	. = ..()
+	. += GLOB.resin_recipes
+
+
+/* pink rune, makes people slightly happier after walking on it*/
+
+
+/obj/item/slimecross/warping/pink
+	colour = "pink"
+	effect_desc = "Draws a rune that makes people happier!"
+	runepath = /obj/effect/warped_rune/pinkspace
+
+
+/obj/effect/warped_rune/pinkspace
+	desc = "Love is the only reliable source of happiness we have left. But like everything, it comes with a price."
+	icon_state = "love_rune"
+
+
+///adds the jolly mood effect along with hug sound effect.
+/obj/effect/warped_rune/pinkspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(istype(crossing,/mob/living/carbon/human))
+		playsound(rune_turf, "sound/weapons/thudswoosh.ogg", 50, TRUE)
+		SEND_SIGNAL(crossing, COMSIG_ADD_MOOD_EVENT,"jolly", /datum/mood_event/jolly)
```

Comment:
can't you just use the hug moodlet (or even the betterhug or besthug moodlets)?

Reply:
I thought about using the hug moodlet but it isn't that high, and the betterhug and besthug moodlet run into a runtime when not used by anyone which it technically isn't since it gives the mood directly without the action

Reply:
can't you just modify the hug and betterhug moodlets to NOT runtime when used/applied by an object or the like?